hack/verify-upgrade.sh

```diff
@@ -0,0 +1,195 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+SCRIPT_ROOT="$( cd "$( dirname "${BASH_SOURCE[0]}" )" > /dev/null && pwd )"
+export REPO_ROOT="${SCRIPT_ROOT}/.."
+source "${REPO_ROOT}/devel/lib/lib.sh"
+source "${REPO_ROOT}/hack/build/version.sh"
+
+kube::version::get_version_vars
+
+LATEST_RELEASE="${KUBE_LAST_RELEASE}"
+CURRENT_VERSION="${KUBE_GIT_VERSION}"
+
+# Ensure helm, kind, kubectl, ytt, jq are available
+bazel build //hack/bin:helm //hack/bin:kind //hack/bin:ytt //hack/bin:jq //hack/bin:kubectl
+bindir="$(bazel info bazel-bin)"
+export PATH="${bindir}/hack/bin/:$PATH"
+
+apply_cm_resources() {
+	selector="$1"
+	# If there is an actual error, this won't work with set -o errexit
+	set +e
```

Comment:
Is `set +e` necessary? Commands that exit != 0 as part of an `if` or `until` don't trigger the `errexit` mechanism, I thought:

`man bash`
> The ERR trap is not executed if the failed command is part of the command list immediately following a while or until 
> keyword, part of the test in an if statement, part of a command executed in a &&  or  ||
>               list  except  the  command following the final && or ||, any command in a pipeline but the last, or if the command's return value is being inverted using !.  These are the same conditions obeyed by the errexit (-e)
>               option.
>

Reply:
Yep, it shouldn't be necessary however in this particular case it seems to not work without that.
I've not figured out why.